PowershellModule/CloudCompute.Powershell/Contracts/WaitableCmdlet.cs

```diff
@@ -13,17 +15,23 @@ public abstract class WaitableCmdlet : PSCmdletCaasWithConnectionBase, IWaitable
 
         protected const int delayTime = 400; 
 
-        protected void WaitForFailureOrCompletion<T>(ResponseType response, Guid objectId)
+        protected void WaitForFailureOrCompletion(ResponseType response, Guid objectId)
         {
             if (this.Wait && response != null && response.responseCode == "IN_PROGRESS")
             {
                 bool provisioned = false;
-                T provisionedObject = default(T);
+                IEntityStatusV2 provisionedObject = default(IEntityStatusV2);
                 
                 while (!provisioned)
                 {
-                    provisioned = WaitOn<T>(objectId, ref provisionedObject);
-                    Thread.Sleep(delayTime);
+                    Update(objectId, ref provisionedObject);
+                    if (provisionedObject.state == "FAILED")
+                        ThrowTerminatingError(
+                           new ErrorRecord(new ComputeApiException(string.Format("Failed to provision {0}", provisionedObject.state)), "-1", ErrorCategory.ConnectionError, Connection));
+
+                    provisioned = (provisionedObject.state != "IN_PROGRESS" && provisionedObject.state != "PENDING_ADD");
```

Comment:
Here is what I think our strategy should be to handle wait,
1> For "**NORMAL**" state, return **success**
2> State starts with "**PENDING_**", then we should **wait** 
3> For Rest all, we should return with **throw** if user has asked us to wait


Some of the possible states are : 
"NORMAL",
"PENDING_ADD",
"PENDING_CHANGE",
"PENDING_DELETE",
"FAILED_ADD",
"FAILED_CHANGE",
"FAILED_DELETE" and
"REQUIRES_SUPPORT".